packages/drivers/routerlicious-driver/src/errorUtils.ts

```diff
@@ -76,43 +77,56 @@ export function createR11sNetworkError(
 			// the error message will start with NetworkError as defined in restWrapper.ts
 			// If there exists a self-signed SSL certificates error, throw a NonRetryableError
 			// TODO: instead of relying on string matching, filter error based on the error code like we do for websocket connections
-			if (errorMessage.includes("failed, reason: self signed certificate")) {
-				return new NonRetryableError(
-					errorMessage,
-					RouterliciousErrorType.sslCertError,
-					props,
-				);
-			}
-			return new GenericNetworkError(
-				errorMessage,
-				errorMessage.startsWith("NetworkError"),
-				props,
-			);
+			error = errorMessage.includes("failed, reason: self signed certificate")
+				? new NonRetryableError(errorMessage, RouterliciousErrorType.sslCertError, props)
+				: new GenericNetworkError(
+						errorMessage,
+						errorMessage.startsWith("NetworkError"),
+						props,
+				  );
+			break;
 		case 401:
 		// The first 401 is manually retried in RouterliciousRestWrapper with a refreshed token,
 		// so we treat repeat 401s the same as 403.
 		case 403:
-			return new AuthorizationError(errorMessage, undefined, undefined, props);
+			error = new AuthorizationError(errorMessage, undefined, undefined, props);
+			break;
 		case 404:
 			const errorType = RouterliciousErrorType.fileNotFoundOrAccessDeniedError;
-			return new NonRetryableError(errorMessage, errorType, props);
+			error = new NonRetryableError(errorMessage, errorType, props);
+			break;
 		case 429:
```

Comment:
Curious, why we are not treating 503 in the same way as 429. Based on my understanding 503 is very similar to 429, the difference is that it is returned by routing layer and not backend service itself.

Reply:
As per the code, if it will have retryAfterMs, then will create same type of error which is handled in default case. @znewton do you have any comment here?

Reply:
I think this is fine as if retryAfterMs is supplied in case of 503, we will create same error as 429. This behaviour is same as odsp error handling.